File: frontend/src/api/partnerOnboarding.ts
import { http } from '@/api/http'
import type { V1Envelope } from '@/api/quotes'

export type PartnerOnboardingStage =
  | 'discovery'
  | 'product_mapping'
  | 'quote_ready'
  | 'portal_ready'
  | 'demo_ready'
  | 'active_partner'
  | 'paused'

export interface PartnerOnboardingChecklistItem {
  key: string
  label: string
  done: boolean
  detail: string
}

export interface PartnerOnboardingLinks {
  partner_detail: string
  product_catalog: string
  demo_walkthrough: string
  market_response: string
  orders: string
}

export interface PartnerOnboardingRecord {
  partner_id: string
  partner_name: string
  partner_code: string | null
  partner_type: string | null
  product_focus: string[]
  target_markets: string[]
  onboarding_stage: PartnerOnboardingStage
  readiness_score: number
  readiness_summary: string
  missing_items: string[]
  next_action: string
  checklist: PartnerOnboardingChecklistItem[]
  links: PartnerOnboardingLinks
  is_reference_partner: boolean
  safety: Record<string, boolean>
}

export interface PartnerOnboardingResponse {
  status: string
  stage_order: string[]
  checklist_keys: string[]
  summary: {
    total_partners: number
    reference_partner_count: number
    demo_ready_count: number
    quote_ready_count: number
    portal_ready_count: number
    active_partner_count: number
    paused_count: number
    safety: Record<string, boolean>
  }
  items: PartnerOnboardingRecord[]
  future_partner_placeholder: PartnerOnboardingRecord | null
  safety: Record<string, boolean>
}

export async function fetchPartnerOnboarding() {
  const { data } = await http.get<V1Envelope<PartnerOnboardingResponse>>('/v1/partner-onboarding')
  return data.data
}

File: backend/app/schemas/partner_onboarding.py
from __future__ import annotations

from typing import Literal
from uuid import UUID

from pydantic import BaseModel


PartnerOnboardingStage = Literal[
    "discovery",
    "product_mapping",
    "quote_ready",
    "portal_ready",
    "demo_ready",
    "active_partner",
    "paused",
]


class PartnerOnboardingChecklistItem(BaseModel):
    key: str
    label: str
    done: bool
    detail: str


class PartnerOnboardingLinks(BaseModel):
    partner_detail: str
    product_catalog: str
    demo_walkthrough: str
    market_response: str
    orders: str


class PartnerOnboardingRecord(BaseModel):
    partner_id: UUID
    partner_name: str
    partner_code: str | None
    partner_type: str | None
    product_focus: list[str]
    target_markets: list[str]
    onboarding_stage: PartnerOnboardingStage
    readiness_score: int
    readiness_summary: str
    missing_items: list[str]
    next_action: str
    checklist: list[PartnerOnboardingChecklistItem]
    links: PartnerOnboardingLinks
    is_reference_partner: bool
    safety: dict[str, bool]


class PartnerOnboardingSummary(BaseModel):
    total_partners: int
    reference_partner_count: int
    demo_ready_count: int
    quote_ready_count: int
    portal_ready_count: int
    active_partner_count: int
    paused_count: int
    safety: dict[str, bool]


class PartnerOnboardingResponse(BaseModel):
    status: str
    stage_order: list[str]
    checklist_keys: list[str]
    summary: PartnerOnboardingSummary
    items: list[PartnerOnboardingRecord]
    future_partner_placeholder: PartnerOnboardingRecord | None
    safety: dict[str, bool]

